src/ethsign/ethsign.go

```diff
@@ -161,19 +157,15 @@ func recover(data []byte, sig hexutil.Bytes, noPrefix bool) (common.Address, err
     hash = signHash(data)
   }
 
-  rpk, err := crypto.Ecrecover(hash, sig)
+  rpk, err := crypto.SigToPub(hash, sig)
   if err != nil {
     return common.Address{}, err
   }
-  pubKey, err := crypto.UnmarshalPubkey(rpk)
-  if err != nil {
-    return common.Address{}, fmt.Errorf("invalid public key")
-  }
-  recoveredAddr := crypto.PubkeyToAddress(*pubKey)
-  return recoveredAddr, nil
+  return crypto.PubkeyToAddress(*rpk), nil
 }
 
 func main() {
+  var defaultHDPath = "m/44'/60'/0'" // aka "ledger legacy"
```

Comment:
What should the default actually be? I'd like to use something more canonical, but changing this would be a breaking change.

Reply:
It would be nice to extend this to also support setting the derivation path scheme by name, e.g. `ethsign --hd-style ledgerlive`. Right now I have to do things like `ethsign ls --hd-path "m/44'/60'/n'/0"` where `n` is the address index (this causes the selected index to be the first address), and I have not been able to find an hd path which corresponds to Ledger Live, i.e. `m/44'/60'/n'` at all. Have you?

Reply:
> Ledger Live uses : m/44'/60'/0'/0
According to this reddit post https://www.reddit.com/r/ledgerwallet/comments/9saceu/confused_at_the_hd_derivation_path_when/e8nvuqo?utm_source=share&utm_medium=web2x